glusterd2/servers/sunrpc/server.go

```diff
@@ -53,6 +55,25 @@ var clientsList = struct {
 func NewMuxed(m cmux.CMux) *SunRPC {
 
 	f := path.Join(config.GetString("rundir"), gd2SocketFile)
+	gd2LockFile := f + ".lock"
+	fd, err := syscall.Open(gd2LockFile,
+		syscall.O_CREAT|syscall.O_WRONLY|syscall.O_CLOEXEC, 0666)
+	if err != nil {
+		log.WithError(err).WithField("lockfile", gd2LockFile).Fatal("failed to open lock file")
+	}
+
+	defer syscall.Close(fd)
```

Comment:
Won't closing unlock the file ? Shouldn't we be holding the lock through-out the lifetime of the process ?

Reply:
I'd prefer if the fd is closed in `Stop()` method after we close the socket file listener. You can store the fd in `SunRPC` structure.

Testcase for this doesn't make much sense and may be really hard even if possible.

However, I'd test this manually by bringing up multiple gd2 instances. One way I could think of:

1. Brick up one gd2 instance and keep it running.
2. Hardcode socket file path to the one used by first instance and recompile gd2
3. Bring up the 2nd instance. This should fail.

Reply:
Fixed